module/Company/src/Company/Controller/AdminController.php

```diff
@@ -508,6 +573,49 @@ public function addCategoryAction()
         ]);
     }
 
+    public function addLabelAction()
+    {
+        // Get useful stuff
+        $companyService = $this->getCompanyService();
+        $labelForm = $companyService->getLabelForm();
+
+        // Handle incoming form results
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            // Check if data is valid, and insert when it is
+            $labels = $companyService->insertLabelByData(
+                $request->getPost(),
+                $request->getFiles()
+            );
+            if (!is_null($labels)) {
+                // Redirect to edit page
+                return $this->redirect()->toRoute(
+                    'admin_company/default',
+                    [
+                        'action' => 'editLabel',
+                        'slugCompanyName' => $labels['nl']->getLanguageNeutralId(),
+                    ]
+                );
+            }
+        }
+
+        // The form was not valid, or we did not get data back
+
+        // Initialize the form
+        $labelForm->setAttribute(
+            'action',
+            $this->url()->fromRoute(
+                'admin_company/default',
+                ['action' => 'addLabel']
+            )
+        );
```

Comment:
Again, looks like work that should not be done here.

Reply:
I tried to be consistent with already existing code in this module.

Reply:
IMO, you should always leave code better than you found it. Never add more trash, especially if the reason is that it was done this way elsewere.

Reply:
I looked into this, and there actually is a reason for this. The same form is being used for two different actions. addLabel and editLabel. The same holds for the Category Form. Moving this piece of code to the form would either result in duplication of code or the Controller still managing this indirectly.